pydantic/_internal/_known_annotated_metadata.py

```diff
@@ -96,82 +139,150 @@ def apply_known_metadata(annotation: Any, schema: CoreSchema) -> CoreSchema | No
         PydanticCustomError: If `Predicate` fails.
     """
     schema = schema.copy()
-    schema_update, _ = collect_known_metadata([annotation])
-    if isinstance(annotation, at.Gt):
-        if schema['type'] in NUMERIC_SCHEMA_TYPES:
-            schema.update(schema_update)  # type: ignore
+    schema_update, other_metadata = collect_known_metadata([annotation])
+    schema_type = schema['type']
+    for constraint, value in schema_update.items():
+        if constraint not in CONSTRAINTS_TO_ALLOWED_SCHEMAS:
+            raise ValueError(f'Unknown constraint {constraint}')
+        allowed_schemas = CONSTRAINTS_TO_ALLOWED_SCHEMAS[constraint]
+
+        if schema_type in allowed_schemas:
+            schema[constraint] = value
+            continue
+
+        if constraint == 'allow_inf_nan' and value is False:
+            return cs.no_info_after_validator_function(
+                _validators.forbid_inf_nan_check,
+                schema,
+            )
+        elif constraint == 'pattern':
+            # insert a str schema to make sure the regex engine matches
+            return cs.chain_schema(
+                [
+                    schema,
+                    cs.str_schema(pattern=value),
+                ]
+            )
```

Comment:
these can probably be made more compact thereby making the function easier to read.

Reply:
black controls formatting not me 😄